File: app/src/main/java/com/bestxty/sault/demo/SingleTaskView.java
package com.bestxty.sault.demo;

import android.content.Context;

/**
 * @author 姜泰阳
 *         Created by 姜泰阳 on 2017/10/19.
 */

public interface SingleTaskView {

    Context getContext();

    void showProgress(long totalSize, long finishedSize);

    void clearProgress();

    void showError(String msg);

    void showStatus(String status);

    void enableStartBtn();

    void enablePauseAndCancelBtn();

    void enableResumeAndCancelBtn();
}

File: app/src/main/java/com/bestxty/sault/demo/SingleTaskView.kt
package com.bestxty.sault.demo

import android.content.Context

/**
 * @author 姜泰阳
 *         Created by 姜泰阳 on 2017/10/19.
 */

interface SingleTaskView {

    fun getContext(): Context

    fun showProgress(totalSize: Long, finishedSize: Long)

    fun clearProgress()

    fun showError(msg: String?)

    fun showStatus(status: String?)

    fun enableStartBtn()

    fun enablePauseAndCancelBtn()

    fun enableResumeAndCancelBtn()
}
